src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Bmi1.PlatformNotSupported.cs

```diff
@@ -17,6 +17,62 @@ public abstract class Bmi1
 
         public static bool IsSupported { get { return false; } }
 
+        public abstract class X64 : Bmi1
```

Comment:
I think it would be better if these inherited from the base `x64` class, rather than the class it is contained in.

If we inherit from the class you are contained in, you have to remember which ISA contains which x64 specific functionality. However, if you instead inherit from the x64 class of the parent, you can see all available x64 specific functions from the "current" ISA (just not any general-purpose functions).

Reply:
Sorry, I missed your previous comment. I think the problem is that some base classes do not have the `x64` class.

Reply:
I think that is probably fine. None of the members are virtual or instanced and we won't generally be adding new APIs to already shipped ISAs (since we will have hopefully fully covered the ISA at time of shipping).

CC. @eerhardt, to see if he has any concerns.

Reply:
Okay, so the `X64` class hierarchy will become

Bmi2.X64 -> Bmi1.X64 -> Object
Lzcnt.X64 -> Object
Popcnt.X64 -> Sse42.X64 -> Sse41.X64 -> Sse2.X64 -> Sse.X64 -> Object

Reply:
Then, the `X64` class cannot access the 32-bit intrinsic of its ISA.
```csahrp
Popcnt.X64.Popcnt(longVal) // OK
Popcnt.X64.Popcnt(intVal)  // error
```

Reply:
Right. And I think that not being able to access the 32-bit intrinsic of the same ISA is fine.

I think that having to remember `Popcnt.x64.Popcnt(longVal)` vs `Popcnt.Popcnt(intVal)` is much easier than having to remember `Sse.x64.ConvertToInt64` vs `Sse41.x64.Extract` (the former is a clear rule of `Isa.x64` for 64-bit specific APIs and `Isa` for everything else; while the latter requires you to remember that `ConvertToInt64` was exposed as part of `Sse.x64` and is not accessible from `Sse41.x64`).

Reply:
I think @tannergooding's proposal makes sense. The good thing is that `Popcnt.X64.Popcnt(intVal)` will be a compile error, and so you will just have to drop `.X64` to make it compile.
I agree that makes sense to only have the `x64`-specific functions available in the `.X64` class. It also gives you the ability to quickly check if you are using any `x64`-specific functions for auditing/reviewing reasons. Calling `Popcnt.X64.Popcnt(intVal)` would actually work on a 32-bit process, so it feels awkward that we allow it to be called through the `.X64` class.

Reply:
@tannergooding @eerhardt Thank you for the comments. Will change.